lib/meilisearch/client.rb

```diff
@@ -10,26 +10,24 @@ def indexes
       http_get '/indexes'
     end
 
-    def show_index(index_uid)
-      index_object(index_uid).show
-    end
-
     # Usage:
     # client.create_index('indexUID')
     # client.create_index('indexUID', primaryKey: 'id')
     def create_index(index_uid, options = {})
       body = options.merge(uid: index_uid)
-      res = http_post '/indexes', body
-      index_object(res['uid'])
+      index_hash = http_post '/indexes', body
+      index_object(index_hash['uid'], index_hash['primaryKey'])
     end
 
     def get_or_create_index(index_uid, options = {})
       begin
-        create_index(index_uid, options)
+        index_instance = fetch_index(index_uid)
```

Comment:
I really prefer this approach, since this method will probably be used more often to retrieve an existing index, than creating an inexistent one 👍